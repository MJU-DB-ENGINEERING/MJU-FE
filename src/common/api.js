
const HOST_URL = 'http://13.125.105.3:50038/';
import { fetch } from 'cross-fetch';

export const doLogin = async (userId, userPw) => {
    const res = await fetch(HOST_URL + 'v1/login', {
        method: "POST",
        headers: {
            "Content-Type": "application/json",
        },
        body: JSON.stringify({
            "userId": userId,
            "userPw": userPw,
        }),
    });
    if (res.status == 200)
        return res.json();
    return undefined;
}

export const searchProject = async (projectName) => {
    const res = await fetch(HOST_URL + 'v1/search/project/' + projectName, {
        method: 'GET',
        headers: {
            "Content-Type": "application/json",
        },
    }).then((response) => response.json());

    return res;
}

export const getProjectDetail = async (projectName) => {
    const res = await fetch(HOST_URL + 'v1/view/' + projectName + '/detail/', {
        method: 'GET',
        headers: {
            "Content-Type": "application/json",
        }
    }).then((response) => response.json());

    return res;
}

export const updateProjectPMEvaluation = async (props) => {
    const { businessComment, businessRate, 
      communicationComment, communicationRate,
      evaluated, evaluator,
      projectId } = props;

    const res = await fetch(HOST_URL + 'v1/evaluation/pm', {
        method: 'POST',
        headers: {
            "Content-Type": "application/json",
        },
        body: JSON.stringify({
            'businessComment': businessComment,
            'businessRate': businessRate,
            'communicationComment': communicationComment,
            'communicationRate': communicationRate,
            'evaluated': evaluated,
            'evaluator': evaluator,
            'projectId': projectId,
        }),
    }).then((response) => response.json());

    return res;
}

export const getProjectPMEvaluation = async (evaluatorName, projectid) => {

    const res = fetch(HOST_URL + 'v1/view/evaluation', {
        method: 'POST',
        headers: {
            "Content-Type": "application/json",
        },
        body: {
            'evaluatorName': evaluatorName,
            'projectid': projectid,
        }
    }).then((res) => res.json());

    return (res);
}