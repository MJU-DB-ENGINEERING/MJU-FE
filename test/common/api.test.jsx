import { assert, describe, it } from 'vitest'
import { doLogin, getProjectDetail, getProjectPMEvaluation, searchProject } from '../../src/common/api';

describe('Login', () => {
  it('Case - Success', async () => {
    const res = await doLogin('breeds50','hands33'); 

    assert.equal(res.userName, '박예연');
  })
});

describe('Search Project', () => {
    it('Case - Success', async () => {
        const res = await searchProject('써머프로젝트2');

        assert.equal(res.id, 7);
        assert.equal(res.projectName, '써머프로젝트2');
    })
})

describe('Get Project Detail', () => {
    it ('Case - Success', async () => {
        const res = await getProjectDetail('써머프로젝트2');
        
        const PM = res.find((e) => e.jobType === 'PM'); 
        assert.equal(PM.employeeName, '박예연');

        const PL = res.find((e) => e.jobType === 'PL'); 
        assert.equal(PL.employeeName, '임예후');

        const Programmer = res.find((e) => e.jobType === 'Programmer'); 
        assert.equal(Programmer.employeeName, '강하준');        

        const Designer = res.find((e) => e.jobType === 'Designer'); 
        assert.equal(Designer.employeeName, '최서아');        

        const Tester = res.find((e) => e.jobType === 'Tester'); 
        assert.equal(Tester.employeeName, '최윤후');    

    });
});

describe('Get Project Evaluations', () => {
    it ('Case - Success', async () => {
        const res = await getProjectPMEvaluation('박예연', 3);

        const e1 = res.find((e) => e.evaluated === '윤윤연'); 
        assert.equal(e1.projectId , 3);
        assert.equal(e1.evaluator, '박예연');

        const e2 = res.find((e) => e.evaluated === '장현윤'); 
        assert.equal(e2.projectId , 3);
        assert.equal(e2.evaluator, '박예연');

        const e3 = res.find((e) => e.evaluated === '조서후'); 
        assert.equal(e3.projectId , 3);
        assert.equal(e3.evaluator, '박예연');

    })
})