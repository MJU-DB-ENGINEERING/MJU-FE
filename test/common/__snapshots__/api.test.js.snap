// Vitest Snapshot v1

exports[`Login > snapshot 1`] = `
{
  "foo": "bar",
}
`;

exports[`Login success test > snapshot 1`] = `
{
  "foo": "bar",
}
`;

exports[`suite name > snapshot 1`] = `
{
  "foo": "bar",
}
`;
